manifests/init.pp

```diff
@@ -358,6 +362,8 @@
   Variant[Undef, Boolean, String] $dhcp_range = undef,
   Optional[String] $dhcp_pxeserver = undef,
   String $dhcp_pxefilename = 'pxelinux.0',
+  String $dhcp_ipxefilename = $foreman_proxy::params::dhcp_ipxefilename,
```

Comment:
I would default to `undef` and determine the value in `proxydhcp.pp` (see the other comment in `params.pp` for details. It also needs to accept `undef` so the data type should be modified:
```suggestion
  Optional[String[1]] $dhcp_ipxefilename = undef,
```